CMakeLists.txt

```
@@ -91,6 +74,56 @@ if(ENABLE_MKLCPU_BACKEND
   list(APPEND DOMAINS_LIST "rng")
 endif()
 
+# Define required CXX compilers before project
+if(CMAKE_CXX_COMPILER)
```

In case compiling with hipSYCL, I believe this part is not necessary, since the selection of the proper compiler will be handled by `syclcc` and the hipSYCL cmake integration. Would it be possible to modify this line like this:

```suggestion
if(CMAKE_CXX_COMPILER OR NOT ONEMKL_SYCL_IMPLEMENTATION STREQUAL "dpc++")
```
Since `ONEMKL_SYCL_IMPLEMENTATION` is set to `dpc++` by default this would not change the behaviour that `dpc++` is selected by default

---

@sbalint98 Thank you for the comments! Could you please review the changes and mark the conversations as resolved after verifying the fix?

---

I think I don't have the permission to mark the conversation resolved... at least I don't see the button for that.

The way cmake works with this change looks good to me, however, I couldn't fully test this PR because #149. Would it be possible to wait until that issue is resolved? If not I can create a local solution for that problem and test with that...

---

@sbalint98 Could you test this PR using a local solution please?

---

Compilation with cmake works as expected with this change!

---

Thank you! I will mark the conversations as resolved.